Tests/TestUtilities/TestRuntime.cs

```diff
@@ -0,0 +1,35 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using NSubstitute;
+
+namespace TestUtilities;
+
+public class TestRuntime : IRuntime
+{
+    public OperatingSystemProvider OperatingSystem { get; init; }
+    public TestLogger Logger { get; init; } = new();
+    public IFileWrapper File { get; init; } = Substitute.For<IFileWrapper>();
+    public IDirectoryWrapper Directory { get; init; } = Substitute.For<IDirectoryWrapper>();
+    ILogger IRuntime.Logger => Logger;
+
+    public TestRuntime() =>
+        OperatingSystem = Substitute.For<OperatingSystemProvider>(File, Logger);
```

Comment:
NIT: My personal preference is to either use field initializers or constructors but to not mix both. If there are dependencies between the fields I use constructors only.